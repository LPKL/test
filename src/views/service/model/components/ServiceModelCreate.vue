<template>
  <div class="app_container">
    <el-dialog
      :visible.sync="dialogVisible"
      :close-on-click-modal="false"
      :close-on-press-escape = "false"
      :show-close="true"
      :title="$t('labels.service.add')"
      class="addDialog"
      @close="closeDialog">
      <el-form ref="createModelServiceForm" :model="createModelServiceForm" :rules="rules" label-width="100px" label-position="right">
        <el-form-item :label="$t('labels.service.name')+':'" prop="name" style="margin-bottom:26px;">
          <el-input v-model="createModelServiceForm.name" clearable style="width:256px;"/>
        </el-form-item>
        <el-form-item :label="$t('labels.desc')+':'" prop="description">
          <el-input :rows="2" v-model="createModelServiceForm.description" type="textarea" style="width:324px;" clearable />
        </el-form-item>
        <el-form-item :label="$t('labels.service.model.index')+':'" prop="model">
          <comp-select-rich
            v-model="createModelServiceForm.model"
            :options="modelsData"
            :option-key="{ value: 'id', label: 'name' }"
            :combind-value="['id','type']"
            :filter-by="filterBy"
            :filterable="true"
            :clearable="true"
            :placeholder="$t('strings.select')"
            class="new-input"/>
        </el-form-item>
        <el-form-item :label="$t('labels.service.model.is_publish')+':'" prop="enabled">
          <el-radio v-model="createModelServiceForm.enabled" :label="true">{{ $t('strings.yes') }}</el-radio>
          <el-radio v-model="createModelServiceForm.enabled" :label="false">{{ $t('strings.no') }}</el-radio>
        </el-form-item>
      </el-form>
      <div slot="footer" class="dialog-footer">
        <el-button class="cancel" @click="closeDialog">{{ $t('buttons.cancel') }}</el-button>
        <el-button type="primary" @click="createNewModelService">{{ $t('buttons.confirm') }}</el-button>
      </div>
    </el-dialog>
  </div>
</template>
<script>

import { getAllPipelineModel } from '@/api/datamining/model'
import SelectRich from '@/components/SelectRich'
import { validateModelServiceName } from '@/utils/validate'

export default {
  name: 'ServiceModelCreate',
  components: {
    'comp-select-rich': SelectRich
  },
  props: {
    isVisable: {
      type: Boolean,
      default: null
    }
  },
  data() {
    const serviceName = (rule, value, callback) => {
      if (!value) {
        callback(new Error(this.$t('rules.required')))
      } else if (!validateModelServiceName(value)) {
        callback(new Error(this.$t('rules.service_name')))
      } else {
        callback()
      }
    }
    return {
      modelsData: [],
      createModelServiceForm: {
        name: '', // 模型名称，必填唯一
        description: '', // 描述
        model: '', // 模型，select，必填
        enabled: true // 是否发布
      },
      rules: {
        name: [
          // { required: true, message: this.$t('rules.required'), trigger: 'blur' },
          { required: true, trigger: 'blur', validator: serviceName }
        ],
        model: [
          { required: true, message: this.$t('rules.required'), trigger: 'blur' }
        ]
      },
      filterBy: { key: 'type', value: [{ label: '单独模型', value: 'model' }, { label: '组合模型', value: 'pipeline' }], single: false }
    }
  },
  computed: {
    dialogVisible: {
      get: function() {
        return this.isVisable
      },
      set: function() {
        //
      }
    }
  },
  watch: {
    dialogVisible(val) {
      if (val === false) {
        this.$refs['createModelServiceForm'].resetFields()
      }
    }
  },
  mounted() {
    this.initData()
  },
  methods: {
    initData(val) {
      getAllPipelineModel().then(res => {
        this.modelsData = res.data
      }).catch(error => {
        console.log('getAllUsersModel error', error)
      })
    },
    closeDialog() {
      this.$emit('closeDialog', 'create')
    },
    createNewModelService() {
      this.$refs['createModelServiceForm'].validate((valid) => {
        if (!valid) return
        const newCreateFrom = {
          name: this.createModelServiceForm.name,
          description: this.createModelServiceForm.description,
          enabled: this.createModelServiceForm.enabled,
          type: this.createModelServiceForm.model.type
        }
        newCreateFrom[this.createModelServiceForm.model.type] = this.createModelServiceForm.model.id
        this.$emit('createdForm', newCreateFrom)
        this.$emit('closeDialog', 'create')
      })
    }
  }

}
</script>
<style rel="stylesheet/scss" lang="scss" scoped>
@import "src/assets/styles/new-base.scss";
.app_container {
  .addDialog {
    /deep/ .el-dialog {
      width: 488px;
      height: 400px;
      border-radius: 16px;
      padding: 24px 24px 24px 24px;
      .el-dialog__header {
        padding-top: 0;
        padding-left: 16px;
        border-bottom: 1px solid #e6e6e6;
        font-size: 18px;
        color: #4d4d4d;
        // font-family: sy_regular;
      }
      .el-dialog__body {
        margin-top: 24px;
        padding: 0 16px 0 16px;
        .el-form {
          .el-form-item {
            margin-bottom: 16px;
            .el-form-item__content {
              .el-textarea {
                padding-top: 10px;
                .el-textarea__inner {
                  height: 64px;
                  width: 324px;
                }
              }
              .el-select {
                margin-top: 0;
                width: 256px;
                .el-input {
                  .el-input__inner {
                    height: 32px;
                    width: 256px;
                    border-radius: 4px;
                  }
                  .el-input__suffix {
                    .el-input__suffix-inner {
                      .el-select__caret {
                        &::before {
                          content: '';
                          position: relative;
                          width: 0;
                          height: 0;
                          top: -10px;
                          border-left: 6px solid transparent;
                          border-right: 6px solid transparent;
                          border-bottom: 6px solid #4d4d4d;
                        }
                      }
                    }
                  }
                }
                // .el-textarea {
                //   .el-textarea__inner {
                //     height: 64px;
                //     width: 324px;
                //   }
                // }
              }
            }
            .el-form-item__label {
              padding-right: 16px;
            }
          }
          .schema_guid,.description {
            .el-form-item__label {
              padding-left: 10px;
            }
          }
          .file_uuid {
            .el-form-item__content {
              .upload-demo {
                .el-upload {
                  .el-button {
                    text-align: left;
                    background-color: #fff;
                    color: #409eff;
                  }
                }
              }
            }
          }
        }
      }
      .el-dialog__footer {
        // margin-top: -10px;
        padding-top: 4px;
        padding-right: 12px;
        .dialog-footer {
          .el-button {
            padding: 0;
            span {
              display: block!important;
              width: 56px;
              height: 32px;
              border-radius: 8px;
              line-height: 32px;
              text-align: center;
            }
          }
          .cancel {
            background-color: #e6e6e6;
            color: #4d4d4d;
            &:hover {
              background-color: #efefef;
            }
            &:active {
              background-color: #ccc;
            }
          }
        }
      }
    }
  }
}
</style>
