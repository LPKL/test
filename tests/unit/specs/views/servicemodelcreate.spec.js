import { mount, createLocalVue } from '@vue/test-utils'
// import index1 from '@/views/service/model/index.vue'
import ServiceModelCreate from '@/views/service/model/components/ServiceModelCreate.vue'
// import index from '@/views/service/model/index.vue'
// import mountOption from '~/cores/vueinstance'

// jest.mock('getAllModel', jest.fn())
import mountOption from '~/cores/vueinstance'

// const $route = {
//   path: '/service/model'
// }

// const createCmp = propsData => mount(ServiceModelCreate, mountOption({ isVisable: true, { stubs: {
//         'form': createModelServiceFormStub,
//       }}, createModelServiceForm: { 'description': '2', 'enabled': true, 'model_id': '3', 'name': '1' }}), { propsData })
const createCmp = propsData => mount(ServiceModelCreate, mountOption({ isVisable: true }), { propsData })

describe('ServiceModelCreate.vue', () => {
  let cmp
  beforeEach(() => {
    // wrapper = mount(ServiceModelCreate, mountOption())
    jest.resetModules()
    jest.clearAllMocks()
  })

  // afterEach(() => {
  //   cmp.destroy()
  // })

  describe('Properties', () => {
    cmp = createCmp()
    it('props default', () => {
      expect(cmp.props().isVisable).toBe(true)
    })

    describe('Validation', () => {
      const props = createCmp().vm.$options.props
      it('prop default type', () => {
        expect(props.isVisable.type).toBe(Boolean)
      })
    })
  })

  describe('Events', () => {
    beforeEach(() => {
      cmp = createCmp({ isVisable: true })
    })

    // method closeDialog
    it('calls closeDialog when chanel button click', () => {
      const mockFn = jest.fn()
      cmp.setMethods({ closeDialog: mockFn })
      const cancelButton = cmp.find('.cancel')
      expect(cancelButton.text()).toBe('取消')
      cmp.find('.cancel').trigger('click')
      // cancelButton.vm.$emit('click')
      // expect(mockFn).toBeCalled()
      // expect(mockFn).toHaveBeenCalledWith(1)
    })

    it('triggers a closeDialog event when a closeDialog method is called', () => {
      const stub = jest.fn()
      cmp.vm.$on('closeDialog', stub)
      cmp.vm.closeDialog()
      expect(stub).toBeCalled()
      expect(stub).toHaveBeenCalledWith('create')
    })

    it('calls closeDialog when @closeDialog happens', () => {
      cmp.find(ServiceModelCreate).vm.$emit('closeDialog', 'create')
      expect(cmp.emitted()['closeDialog'][0]).toEqual(['create'])
    })

    // method createNewModelService
    it('calls handleClick when click on message', () => {
      const createNewModelService = jest.fn()
      cmp.setMethods({ createNewModelService })
      expect(cmp.find('.el-button--primary').text()).toBe('确定')
      // cmp.find('.el-button--primary').trigger('click')
      // expect(createNewModelService).toBeCalled()
      // expect(createNewModelService).toHaveBeenCalledWith(1)
    })

    it('triggers a message-clicked event when a handleClick method is called', async() => {
      const stub1 = jest.fn()
      cmp.setData({ createModelServiceForm: { 'description': '2', 'enabled': true, 'model_id': '3', 'name': '1' }})
      cmp.vm.$on('createdForm', stub1)
      cmp.vm.createNewModelService()
      // await cmp.vm.$refs.createModelServiceForm.validate()
      // expect(cmp.vm.$refs.createModelServiceForm.validate.errors.any()).toBe(true)
      expect(stub1).toBeCalled()
      expect(stub1).toHaveBeenCalledWith({ 'description': '2', 'enabled': true, 'model_id': '3', 'name': '1' })
    })
  })

  it('snapshot', () => {
    expect(cmp.element).toMatchSnapshot()
    // expect(wrapper.html()).toMatchSnapshot()
  })
})

