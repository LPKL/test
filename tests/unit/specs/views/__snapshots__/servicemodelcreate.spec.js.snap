// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceModelCreate.vue snapshot 1`] = `
<div
  class="app_container"
>
  <div
    class="el-dialog__wrapper addDialog"
    style="z-index: 2013;"
  >
    <div
      aria-label="添加服务"
      aria-modal="true"
      class="el-dialog"
      role="dialog"
      style="margin-top: 15vh;"
    >
      <div
        class="el-dialog__header"
      >
        <span
          class="el-dialog__title"
        >
          添加服务
        </span>
        <!---->
      </div>
      <div
        class="el-dialog__body"
      >
        <form
          class="el-form el-form--label-right"
        >
          <div
            class="el-form-item is-success is-required"
          >
            <label
              class="el-form-item__label"
              for="name"
              style="width: 100px;"
            >
              服务名称
            </label>
            <div
              class="el-form-item__content"
              style="margin-left: 100px;"
            >
              <div
                class="el-input el-input--suffix"
                style="width: 256px;"
              >
                <!---->
                <input
                  autocomplete="off"
                  class="el-input__inner"
                  type="text"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
           
          <div
            class="el-form-item"
          >
            <label
              class="el-form-item__label"
              for="description"
              style="width: 100px;"
            >
              描述
            </label>
            <div
              class="el-form-item__content"
              style="margin-left: 100px;"
            >
              <div
                class="el-textarea el-input--suffix"
                style="width: 324px;"
              >
                <textarea
                  autocomplete="off"
                  class="el-textarea__inner"
                  rows="2"
                  style="min-height: -4px;"
                />
                <!---->
              </div>
              <!---->
            </div>
          </div>
           
          <div
            class="el-form-item is-success is-required"
          >
            <label
              class="el-form-item__label"
              for="model_id"
              style="width: 100px;"
            >
              模型
            </label>
            <div
              class="el-form-item__content"
              style="margin-left: 100px;"
            >
              <div
                class="el-select new-input"
                placeholde="请选择"
              >
                <!---->
                <div
                  class="el-input el-input--suffix"
                >
                  <!---->
                  <input
                    autocomplete="off"
                    class="el-input__inner"
                    placeholder="请选择"
                    readonly="readonly"
                    type="text"
                  />
                  <!---->
                  <span
                    class="el-input__suffix"
                  >
                    <span
                      class="el-input__suffix-inner"
                    >
                      <i
                        class="el-select__caret el-input__icon el-icon-arrow-up"
                      />
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                    </span>
                    <i
                      class="el-input__icon el-input__validateIcon el-icon-circle-check"
                    />
                  </span>
                  <!---->
                  <!---->
                </div>
                <div
                  class="el-select-dropdown el-popper"
                  style="display: none;"
                >
                  <div
                    class="el-scrollbar"
                    style="display: none;"
                  >
                    <div
                      class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
                    >
                      <ul
                        class="el-scrollbar__view el-select-dropdown__list"
                      />
                    </div>
                    <div
                      class="el-scrollbar__bar is-horizontal"
                    >
                      <div
                        class="el-scrollbar__thumb"
                        style="transform: translateX(0%); webkit-transform: translateX(0%);"
                      />
                    </div>
                    <div
                      class="el-scrollbar__bar is-vertical"
                    >
                      <div
                        class="el-scrollbar__thumb"
                        style="transform: translateY(0%); webkit-transform: translateY(0%);"
                      />
                    </div>
                  </div>
                  <p
                    class="el-select-dropdown__empty"
                  >
                    
          无数据
        
                  </p>
                </div>
              </div>
              <!---->
            </div>
          </div>
           
          <div
            class="el-form-item"
          >
            <label
              class="el-form-item__label"
              for="enabled"
              style="width: 100px;"
            >
              是否发布
            </label>
            <div
              class="el-form-item__content"
              style="margin-left: 100px;"
            >
              <label
                aria-checked="true"
                class="el-radio is-checked"
                role="radio"
                tabindex="0"
              >
                <span
                  class="el-radio__input is-checked"
                >
                  <span
                    class="el-radio__inner"
                  />
                  <input
                    aria-hidden="true"
                    class="el-radio__original"
                    tabindex="-1"
                    type="radio"
                    value="true"
                  />
                </span>
                <span
                  class="el-radio__label"
                >
                  是
                  <!---->
                </span>
              </label>
               
              <label
                class="el-radio"
                role="radio"
                tabindex="0"
              >
                <span
                  class="el-radio__input"
                >
                  <span
                    class="el-radio__inner"
                  />
                  <input
                    aria-hidden="true"
                    class="el-radio__original"
                    tabindex="-1"
                    type="radio"
                    value="false"
                  />
                </span>
                <span
                  class="el-radio__label"
                >
                  否
                  <!---->
                </span>
              </label>
              <!---->
            </div>
          </div>
        </form>
         
      </div>
      <div
        class="el-dialog__footer"
      >
        <div
          class="dialog-footer"
        >
          <button
            class="el-button cancel el-button--default"
            type="button"
          >
            <!---->
            <!---->
            <span>
              取消
            </span>
          </button>
           
          <button
            class="el-button el-button--primary"
            type="button"
          >
            <!---->
            <!---->
            <span>
              确定
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
